src/EntityId/PrefixMappingEntityIdParser.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Wikibase\DataModel\Services\EntityId;
+
+use Wikimedia\Assert\Assert;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\EntityIdParser;
+
+/**
+ * EntityIdParser that adds a fixed prefix and parses resulting string as an EntityId.
+ * This can be used to prefix IDs of entities coming from a foreign repository
+ * with the repository name to avoid clashes with IDs of local entities.
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class PrefixMappingEntityIdParser implements EntityIdParser {
+
+	/**
+	 * @var string
+	 */
+	private $prefix;
+
+	/**
+	 * @var string[]
+	 */
+	private $prefixMapping;
+
+	/**
+	 * @var EntityIdParser
+	 */
+	private $idParser;
+
+	/**
+	 * @param string $prefix Prefix to be added. It should not contain a colon, in particular at the end of the prefix
+	 * @param string[] $prefixMapping
+	 * @param EntityIdParser $idParser
+	 */
+	public function __construct( $prefix, array $prefixMapping, EntityIdParser $idParser ) {
+		Assert::parameterType( 'string', $prefix, '$prefix' );
+		Assert::parameter( strpos( $prefix, ':' ) === false, '$prefix', 'must not contain a colon' );
```

Comment:
Can you add the relevant @throws docs to the constructor? We'd be doing this in case of inline exception throwing, and where it's definitely more valuable since it's a lot less easy to see what's happening otherwise.